Sources/edge/Commands/Devices+linux.swift

```diff
@@ -0,0 +1,95 @@
+#if os(Linux)
+    import Foundation
+    import Logging
+
+    struct PlatformDeviceDiscovery: DeviceDiscovery {
+        func listUSBDevices(logger: Logger) {
+            logger.info("Listing USB devices on Linux")
+
+            var foundEdgeOSDevices = false
+
+            // Use /sys/bus/usb/devices to list USB devices
+            let task = Process()
+            task.executableURL = URL(fileURLWithPath: "/bin/bash")
+            task.arguments = ["-c", "lsusb"]
+
+            let pipe = Pipe()
+            task.standardOutput = pipe
+
+            do {
+                try task.run()
+                let data = pipe.fileHandleForReading.readDataToEndOfFile()
+                if let output = String(data: data, encoding: .utf8) {
+                    for line in output.split(separator: "\n") {
+                        let deviceInfo = String(line)
+                        logger.debug("Found USB device: \(deviceInfo)")
+
+                        // Filter for EdgeOS devices
+                        if deviceInfo.contains("EdgeOS") {
+                            print(deviceInfo)
+                            foundEdgeOSDevices = true
+                        }
+                    }
+                }
+                task.waitUntilExit()
+            } catch {
+                logger.error("Failed to list USB devices: \(error)")
+            }
+
+            if !foundEdgeOSDevices {
+                logger.info("No EdgeOS USB devices found")
+            }
+        }
+
+        func listEthernetInterfaces(logger: Logger) {
+            logger.info("Listing Ethernet interfaces on Linux")
+
+            var foundEdgeOSInterfaces = false
+
+            // Use ip link show to list network interfaces
+            let task = Process()
+            task.executableURL = URL(fileURLWithPath: "/bin/bash")
+            task.arguments = ["-c", "ip link show"]
```

Comment:
I don't think you're supposed to parse the output of `ip link show`. But you can use `ip link -j show` instead. Using `-j` you'll get JSON output, which you can then parse using `JSONDecoder`.

Reply:
what does "supposed to"? it prints output, I should be able to do whatever I want with the output. I did run into a problem because foundation can't actually parse it because the output starts the same way as stack trace output so foundation gets confused. good call on the -j. I'll use it.